esi/lease/v1/_proxy.py

```diff
@@ -0,0 +1,240 @@
+                specified ``available_start_time`` or an empty set if not found.
+            * ``available_end_time``:  Only return those with the
+              specified ``available_start_time`` or an empty set if not found.
+            * ``project_id``:only return the ones associated with this specific
+              project UUID.
+            * ``node_type``:only return the ones associated with this specific
+              node type.
+            * ``resource_uuid``:only return the ones associated with this specific
+              resource uuid.
+            * ``resource_class``:only return the ones associated with this specific
+              resource class.
+
+        :returns: A generator of offer instances.
+        """
+        return _offer.Offer.list(self, details=details, **query)
+
+    def create_offer(self, **attrs):
+        """Create a new offer from attributes.
+
+        :param dict attrs: Keyword arguments that will be used to create a
+            :class:`~esi_leap.v1.offer.Offer`.
+
+        :returns: The results of offer creation.
+        :rtype: :class:`~esi_leap.v1.offer.Offer`.
+        """
+        return self._create(_offer.Offer, **attrs)
+
+    def get_offer(self, offer, fields=None):
+        """Get a specific offer.
+
+        :param offer: The value can be the ID of an offer or a
+            :class:`~esi_leap.v1.offer.Offer` instance.
+        :param fields: Limit the resource fields to fetch.
+
+        :returns: One :class:`~esi_leap.v1.offer.Offer`
+        :raises: :class:`~openstack.exceptions.ResourceNotFound` when no
+            offer matching the name or ID could be found.
+        """
+        return self._get_with_fields(_offer.Offer, offer, fields=fields)
+
+    def delete_offer(self, offer, ignore_missing=True):
+        """Delete an offer.
+
+        :param offer: The value can be either the ID of an offer or
+            a :class:`~esi_leap.v1.offer.Offer` instance.
+        :param bool ignore_missing: When set to ``False``, an exception
+            :class:`~openstack.exceptions.ResourceNotFound` will be raised
+            when the offer could not be found. When set to ``True``, no
+            exception will be raised when attempting to delete a non-existent
+            offer.
+
+        :returns: The instance of the offer which was deleted.
+        :rtype: :class:`~esi_leap.v1.offer.Offer`.
+        """
+        return self._delete(_offer.Offer, offer, ignore_missing=ignore_missing)
+
+    def claim_offer(self, offer, **attrs):
+        """Claim an offer.
+
+        :param offer: Either the ID of a offer or an instance
+            of :class:``~esi_leap.v1.offer.Offer`.
+        :param dict attrs: The attributes to update on the offer represented
+            by the ``offer`` parameter.
+
+        :returns: The claimed offer.
+        :rtype: :class:`~esi.v1.offer.Offer`
+        """
+        return self._update(_offer.Offer, offer, **attrs)
```

Comment:
A claim isn't really an update; it ends up going to a custom URL (https://github.com/CCI-MOC/python-esileapclient/blob/master/esileapclient/v1/offer.py#L117). Does this accommodate that?

Reply:
I think creating another resource class OfferClaim in esi/lease/v1/offer.py and set the base_path value to be /offers/%(offer_id)s/claim could work.